src/role_sourcer.js

```diff
@@ -43,6 +43,9 @@ roles.sourcer.killPrevious = true;
 roles.sourcer.flee = false;
 
 roles.sourcer.preMove = function(creep, directions) {
+  if (creep.allowOverTake(directions)) {
```

Comment:
This is preventing my sourcer from moving at all in some situations, notably when there's another sourcer "behind" it on the same path.

Reply:
it shouldn't : https://github.com/TooAngel/screeps/pull/415/files
did some test from 0 to spawn move without pb. But you can always modify moveCreep for fix if it's really comming from here.

Reply:
also, if you used your oposite drection util fct, it's normal as i's not the oposite but the oposite  +/- 3

Reply:
To be specific, this makes my sourcers get stuck when they are moving on the same path, one behind the other, across swamp with no road.

Reply:
The thing is that i didn't add sourcers as exeption because in my codebase, i use MW as layoutString for external sourcers so i allowed my external sourcers to overTake internal ones.
For add them as exeption, just add it as i did for harvesters in the line i mentionned. I could do it but i prefer change sourcers external layoutString if i'll gonna look again into it ^^'.
Ofc, the moveCreep method is clearly not finish and there are some work to do on. I did that quickly because i had beautifull lines of creeps following sourcers .
Which are the others situations you notice else ?

Reply:
The situation where I have a problem is two sourcers trying to cross a swamp. The lead sourcer stops for the rear sourcer to overtake, and the rear sourcer never overtakes.

Reply:
did you merged #426 ? how did you solved conflict. may be the cause . as i tried it again from master and sourcer overtaken corectly another one (first was stuck until second pass)